### PR TITLE
Fix StyleCop Issues - Document Parameters

### PR DESCRIPTION
**Pull request by mark-monteiro**
**This builds on PR #23 and should not be reviewed/merged until that one has been merged**

### CODE REVIEW

**Comment by Bond-009** [comment]
@mark-monteiro still a draft?

**Comment by mark-monteiro** [comment]
> @mark-monteiro still a draft?

No, was just waiting for the previous PR to get merged so this didn't get reviewed/merged out-of order

File: Emby.AutoOrganize/Api/FileOrganizationService.cs [hunk 1]
```diff
@@ -1,3 +1,6 @@
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+#pragma warning disable SA1600 // Elements should be documented
```
Comment (Bond-009) [inline Emby.AutoOrganize/Api/FileOrganizationService.cs]: We disabled this in the main project

File: Emby.AutoOrganize/Data/BaseSqliteRepository.cs [hunk 2]
```diff
@@ -1,9 +1,13 @@
+// TODO: This class should be replaced with the common class in the main Jellyfin repository:
+// https://github.com/jellyfin/jellyfin/blob/master/Emby.Server.Implementations/Data/BaseSqliteRepository.cs
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+#pragma warning disable SA1600 // Elements should be documented
```
Comment (Bond-009) [inline Emby.AutoOrganize/Data/BaseSqliteRepository.cs]: Same

File: Emby.AutoOrganize/Data/ManagedConnection.cs [hunk 3]
```diff
@@ -1,3 +1,8 @@
+// TODO: This class should be replaced with the common class in the main Jellyfin repository:
+// https://github.com/jellyfin/jellyfin/blob/master/Emby.Server.Implementations/Data/ManagedConnection.cs
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+#pragma warning disable SA1600 // Elements should be documented
```
Comment (Bond-009) [inline Emby.AutoOrganize/Data/ManagedConnection.cs]: Same

File: Emby.AutoOrganize/Data/SqliteExtensions.cs [hunk 4]
```diff
@@ -1,15 +1,63 @@
+// TODO: This class should be replaced with the common class in the main Jellyfin repository:
+// https://github.com/jellyfin/jellyfin/blob/master/Emby.Server.Implementations/Data/SqliteExtensions.cs
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+#pragma warning disable SA1600 // Elements should be documented
```
Comment (Bond-009) [inline Emby.AutoOrganize/Data/SqliteExtensions.cs]: Same

File: Emby.AutoOrganize/Emby.AutoOrganize.csproj [hunk 5]
```diff
@@ -4,6 +4,11 @@
     <TargetFramework>netstandard2.0</TargetFramework>
     <AssemblyVersion>5.0.0</AssemblyVersion>
     <FileVersion>5.0.0</FileVersion>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```
Comment (Bond-009) [inline Emby.AutoOrganize/Emby.AutoOrganize.csproj:11]: This can be made one line

File: Emby.AutoOrganize/Model/SmartMatchInfo.cs [hunk 6]
```diff
@@ -1,20 +1,27 @@
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+#pragma warning disable SA1600 // Elements should be documented
```
Comment (Bond-009) [inline Emby.AutoOrganize/Model/SmartMatchInfo.cs]: Same

File: Emby.AutoOrganize/PluginEntryPoint.cs [hunk 7]
```diff
@@ -71,13 +84,13 @@ public Task RunAsync()
             Current = this;
             FileOrganizationService = new FileOrganizationService(_taskManager, _repository, _logger, _libraryMonitor, _libraryManager, _config, _fileSystem, _providerManager);
 
-            FileOrganizationService.ItemAdded += _organizationService_ItemAdded;
-            FileOrganizationService.ItemRemoved += _organizationService_ItemRemoved;
-            FileOrganizationService.ItemUpdated += _organizationService_ItemUpdated;
-            FileOrganizationService.LogReset += _organizationService_LogReset;
+            FileOrganizationService.ItemAdded += OrganizationService_ItemAdded;
```
Comment (Bond-009) [inline Emby.AutoOrganize/PluginEntryPoint.cs]:
```suggestion
            FileOrganizationService.ItemAdded += OnOrganizationServiceItemAdded;
```
;)

**Review by Bond-009 — CHANGES_REQUESTED** [review]
Finally found the time :)

File: Emby.AutoOrganize/Emby.AutoOrganize.csproj (hunk 5 above)
Comment (mark-monteiro) [reply Emby.AutoOrganize/Emby.AutoOrganize.csproj:11]: How would you do that? I tried this but then the project wouldn't load:

```
<PropertyGroup Condition=" '$(Configuration)' == 'Release' " TreatWarningsAsErrors="true"/>
```

> error  : The attribute "TreatWarningsAsErrors" in element <PropertyGroup> is unrecognized.

Comment (Bond-009) [reply Emby.AutoOrganize/Emby.AutoOrganize.csproj:11]:
```xml
    <TreatWarningsAsErrors Condition=" '$(Configuration)' == 'Release' ">true</TreatWarningsAsErrors>
```
And put it in the `PropertyGroup` above

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**